profiles/utils.py

```diff
@@ -132,53 +133,68 @@ def generate_filepath(filename, username, suffix=''):
         name=name[:max_name_length],
         timestamp=timestamp.strftime("%Y-%m-%dT%H%M%S"),
         prefix=IMAGE_PATH_PREFIX,
-        username=username,
+        directory_name=directory_name,
         suffix=suffix,
         ext=ext,
     )
 
     return full_path
 
 
-def _generate_upload_to_uri(suffix=""):
+def _generate_upload_to_uri(suffix, directory_getter):
     """
     Returns a function to specify the upload directory and filename, via upload_to on an ImageField
 
     Args:
         suffix (str):
             A suffix for the filename
+        directory_getter (function): A function to get the directory name, instance => directory_name
     Returns:
         function:
             A function to use with upload_to to specify an upload directory and filename
     """
 
     def _upload_to(instance, filename):
         """Function passed to upload_to on an ImageField"""
-        return generate_filepath(filename, instance.user.username, suffix)
+        return generate_filepath(filename, directory_getter(instance), suffix)
 
     return _upload_to
```

Comment:
I feel like this is getting too complex now, we're not passing in a `getter` for `directory_name` because we're return a function which we invoke immediately.

I think we should instead remove this `_generate_upload_to_uri` function altogether and instead directly call `generate_filepath` in the other 5 functions